### PR TITLE
Review

### PR DESCRIPTION
**Pull request by polinakorotenkova**

### CODE REVIEW

File: api.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+  if (request.method == 'POST') {
+    if (request.url == '/login') {
+      let body = '';
+      request.on('data', chunk => {
+        body += chunk.toString();
+      });
+      request.on('end', async () => {
+        const data = JSON.parse(body)
+        let result = await loginQuery(data.email, data.password)
```
Comment (LevaSokolov) [inline api.js]: const

File: api.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        const result1 = (JSON.stringify({ error: 'invalid login or password' }))
+        if (!result) {
+          response.statusCode = 401;
+          response.end(result1)
+
+          return;
+        }
+        var token = jwt.sign({ userId: result.id,name: data.name }, 'nasrat');
```
Comment (LevaSokolov) [inline api.js]: const

File: api.js [hunk 3]
```diff
@@ -0,0 +1,157 @@
+const { error } = require('console');
+const http = require('http');
+var jwt = require('jsonwebtoken');
```
Comment (LevaSokolov) [inline api.js]: const

File: api.js [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+    }
+
+    if (request.url == '/registration') {
+      let body = '';
+      request.on('data', chunk => {
+        body += chunk.toString();
+      });
+      request.on('end', async () => {
+        const data = JSON.parse(body)
+        let result = await regisration(data.email, data.password, data.name)
```
Comment (LevaSokolov) [inline api.js]: const

File: api.js [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+        console.log(result.rows[0].id)
```
Comment (LevaSokolov) [inline api.js]: delete console log

File: api.js [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+
+        if (!result) {
+          const result1 = (JSON.stringify({ error: 'this login is already in use' }));
+          response.statusCode = 401;
+          response.end(result1)
+          return
+        }
+
+        var jwt = require('jsonwebtoken');
+        var token = jwt.sign({ userId: result.rows[0].id ,name: data.name, }, 'nasrat');
```
Comment (LevaSokolov) [inline api.js]: const

File: api.js [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+    if (request.url == '/addtodos') {
+      let body = '';
+      request.on('data', chunk => {
+        body += chunk.toString();
+      });
+      request.on('end', async () => {
+        const data = JSON.parse(body)
+        const token = request.headers.authorization;
+        const decoded = jwt.verify(token, 'nasrat');
+        let result = await addTodos(data.text, data.isDone, decoded.userId)
```
Comment (LevaSokolov) [inline api.js]: const

File: api.js [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+        }
+        response.end(result2)})
+    }
+  }
+
+  if (request.method == 'GET') {
+    if (request.url == '/todos') {
+        const token = request.headers.authorization;
+        const decoded = jwt.verify(token, 'nasrat');
+        let result = await receivingTodos(decoded.userId)
```
Comment (LevaSokolov) [inline api.js]: const

File: api.js [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+      let body = '';
+      request.on('data', chunk => {
+        body += chunk.toString();
+      });
+      request.on('end', async () => {
+        const data = JSON.parse(body)
+        const token = request.headers.authorization;
+        const decoded = jwt.verify(token, 'nasrat');
+        const userId = decoded.userId
+        let result = await deleteTodos(data.id, userId)
```
Comment (LevaSokolov) [inline api.js]: const

File: api.js [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+      let body = '';
+      request.on('data', chunk => {
+        body += chunk.toString();
+      });
+      request.on('end', async () => {
+        const data = JSON.parse(body)
+        const token = request.headers.authorization;
+        const decoded = jwt.verify(token, 'nasrat');
+        const userId = decoded.userId
+        let result = await changeTodo (data.id, userId, data.text, data.isDone)
```
Comment (LevaSokolov) [inline api.js]: const

File: api.js [hunk 11]
```diff
@@ -0,0 +1,157 @@
...
+          response.statusCode = 401;
+          response.end(result1)
+          return
+        }
+        response.end(result2)
+      })
+    }
+  }
+
+  console.log(`Запрошенный адрес: ${request.url}`);
```
Comment (LevaSokolov) [inline api.js]: delete

File: api.js [hunk 12]
```diff
@@ -0,0 +1,157 @@
...
+
+}).listen(2000, "127.0.0.1", async function () {
+  console.log('подключаемся к базе данных')
```
Comment (LevaSokolov) [inline api.js]: delete console log

File: api.js [hunk 13]
```diff
@@ -0,0 +1,157 @@
...
+  await client.connect()
+  console.log('Сервер начал прослушивание запросов на порту 2000');
```
Comment (LevaSokolov) [inline api.js]: delete console log

File: changeTodos.js [hunk 14]
```diff
@@ -0,0 +1,25 @@
+const { client } = require('./connect')
+
+async function changeTodo(id, userId, text, isDone) {
+  
+  let textSql = ''
+  let isDoneSql = ''
```
Comment (LevaSokolov) [inline changeTodos.js]: const

File: api.js [hunk 15]
```diff
@@ -0,0 +1,157 @@
...
+      });
+    }
+
+    if (request.url == '/addtodos') {
+      let body = '';
+      request.on('data', chunk => {
+        body += chunk.toString();
+      });
+      request.on('end', async () => {
+        const data = JSON.parse(body)
```
Comment (MusicCorner) [inline api.js]: move into a separated function

**Review by MusicCorner — CHANGES_REQUESTED** [review]
Install eslint to you project + install eslint extension if you’re using vscode (use airbnb rules)

Move strings, numbers and all that stuff into constants

Also please move repeatable code into functions

**Pull request merged**